health-services/individual/src/main/java/org/egov/individual/repository/IndividualRepository.java

```diff
@@ -24,4 +37,146 @@ protected IndividualRepository(Producer producer,
+                                 String tenantId, Long lastChangedSince, Boolean includeDeleted) {
+        Map<String, Object> paramsMap = new HashMap<>();
+        String query = getQueryForIndividual(searchObject, limit, offset, tenantId, lastChangedSince,
+                includeDeleted, paramsMap);
+        if (searchObject.getIdentifier() == null) {
+            List<Individual> individuals = this.namedParameterJdbcTemplate.query(query, paramsMap, this.rowMapper);
+            if (!individuals.isEmpty()) {
+                enrichIndividuals(individuals);
+            }
+            return individuals;
+        } else {
+            Map<String, Object> identifierParamMap = new HashMap<>();
+            String identifierQuery = getIdentifierQuery(searchObject.getIdentifier(), identifierParamMap);
+            List<Identifier> identifiers = this.namedParameterJdbcTemplate
+                    .query(identifierQuery, identifierParamMap, new IdentifierRowMapper());
+            if (!identifiers.isEmpty()) {
+                query = query.replace(" AND tenantId=:tenantId ", " AND tenantId=:tenantId AND id=:individualId ");
+                paramsMap.put("individualId", identifiers.stream().findAny().get().getIndividualId());
+                List<Individual> individuals = this.namedParameterJdbcTemplate.query(query,
+                        paramsMap, this.rowMapper);
+                if (!individuals.isEmpty()) {
+                    individuals.forEach(individual -> {
+                        individual.setIdentifiers(identifiers);
+                        List<Address> addresses = getAddressForIndividual(individual.getId());
+                        individual.setAddress(addresses);
+                    });
+                }
+                return individuals;
+            }
+            return Collections.emptyList();
+        }
+    }
+
+    private String getQueryForIndividual(IndividualSearch searchObject, Integer limit, Integer offset, String tenantId, Long lastChangedSince, Boolean includeDeleted, Map<String, Object> paramsMap) {
+        String query = "SELECT * FROM individual";
+        List<String> whereFields = GenericQueryBuilder.getFieldsWithCondition(searchObject, QueryFieldChecker.isNotNull, paramsMap);
+        query = GenericQueryBuilder.generateQuery(query, whereFields).toString();
+
+        query += " AND tenantId=:tenantId ";
+        if (query.contains(tableName + " AND")) {
+            query = query.replace(tableName + " AND", tableName + " WHERE ");
+        }
+        if (searchObject.getGender() != null) {
+            query = query + "AND gender =:gender ";
+            paramsMap.put("gender", searchObject.getGender().name());
+        }
+        if (searchObject.getDateOfBirth() != null) {
+            query = query + "AND dateOfBirth =:dateOfBirth ";
+            paramsMap.put("dateOfBirth", Date.valueOf(searchObject.getDateOfBirth()));
+        }
+        if (Boolean.FALSE.equals(includeDeleted)) {
```

Comment:
shouldn't we be comparing with TRUE i.e. `includeDeleted` is `true` means include deleted entities in response. API spec has `includeDeleted` default as `false` so if false or not passed, deleted entities should not be sent.